include/odp/api/spec/ipsec.h

```diff
@@ -937,6 +937,9 @@ typedef struct odp_ipsec_stats_t {
 
 	/** Number of packets with hard lifetime(packets) expired */
 	uint64_t hard_exp_pkts_err;
+
+	/** Bytes of data processed successfully */
+	uint64_t success_bytes;
```

Comment:
I wonder if this is too vague. The "of data" part sounds like it really means something, although I am not sure what that would be.

Maybe this could say: "The packet bytes included in the count are the same as in the byte based lifetime calculation." But it would not be much better since lifetime config does not specify the exact bytes either (and maybe rightly so). And maybe some implementations cannot easily make these two byte counts the same.

Or perhaps this could say: "The range of packet bytes included in the count is implementation defined but includes at least the bytes encrypted or decrypted in ESP or the bytes authenticated in AH". This would appear to match the linux-gen implementation.

Or maybe simply: "The range of packet bytes included in the count is implementation defined."

Depending on if and how the API text gets adjusted, the validation test could be made to check more than just that the count is greater than zero.

Reply:
Let me know if the following text looks fine:

/** Total bytes of packet data processed by IPsec SA  in success cases
 *
 * The range of packet bytes included in the success_bytes count is implementation defined but includes
 * at least the bytes input for encryption or bytes output after decryption in ESP or the bytes authenticated in AH.
 */

Will update the validation test to check accordingly in the next version.

Reply:
How about: ...at least the encrypted bytes in ESP and the authenticated bytes in AH.

Reply:
The wording "at least encrypted bytes" would mean that implementation would need to account for padded bytes. This may not be supported on all platforms.

My intention was to specify the minimum value as the data range in the plain packet being encrypted in case of encryption. In case of decryption, the final output data range which will be placed in the resulting packet. This would keep the padding out of the equation.

Reply:
I see. What about TFC padding in that case? Undefined whether it gets included at all)? Maybe that would be fine and if you think so too, I am ok with the wording you suggested. In that case you can already add my review tag in the next version.

Reply:
I think better to leave the TFC padding part unspecified, as it would allow flexibility to the implementations. And we are only looking at defining the minimum value to be counted.

> I am ok with the wording you suggested. In that case you can already add my review tag in the next version.

@JannePeltonen shall I add reviewed-by tag to linux-generic commit as well?

Reply:
> shall I add reviewed-by tag to linux-generic commit as well?

Yes, you can do so.